static/js/publisher/release/actions/releases.test.js

```diff
@@ -12,4 +29,234 @@ describe("releases actions", () => {
+      expect(getErrorMessage({})).toEqual(ERROR_MESSAGE);
+    });
+
+    it("should return error.message if defined", () => {
+      expect(getErrorMessage({ message: "error!" })).toEqual("error!");
+    });
+
+    it("should return multiple messages if errors.json is an array", () => {
+      expect(
+        getErrorMessage({
+          message: "error!",
+          json: [{ message: "error1" }, { message: "error2" }]
+        })
+      ).toEqual("error! error1 error2");
+    });
+
+    it("should return multiple message if errors.json.errors is an array", () => {
+      expect(
+        getErrorMessage({
+          message: "error!",
+          json: { errors: [{ message: "error1" }, { message: "error2" }] }
+        })
+      ).toEqual("error! error1 error2");
+    });
+  });
+
+  describe("handleReleaseResponse", () => {
+    it("should dispatch all the actions", () => {
+      const store = mockStore({});
+
+      const dispatch = store.dispatch;
+      const json = {
+        success: true,
+        channel_map_tree: {
+          latest: {
+            16: {
+              amd64: [
+                {
+                  channel: "edge",
+                  info: "specific",
+                  revision: 3,
+                  version: "test"
+                }
+              ]
+            }
+          }
+        }
+      };
+
+      const release = [
+        {
+          id: 3,
+          revision: 3,
+          channels: ["latest/edge"]
+        }
+      ];
+
+      const revision = {
+        architectures: ["amd64"],
+        revision: 3
+      };
+
+      const revisions = {
+        "3": revision
+      };
+
+      handleReleaseResponse(dispatch, json, release, revisions, "latest");
+
+      const actions = store.getActions();
+
+      expect(actions[0]).toEqual({
+        payload: {
+          channel: "latest/edge",
+          revision: revision
+        },
+        type: "RELEASE_REVISION_SUCCESS"
+      });
+    });
+  });
+
+  describe("releaseRevisions", () => {
+    beforeEach(() => {
+      global.fetch = jest.fn().mockResolvedValue();
+    });
+    afterEach(() => {
+      global.fetch.mockRestore();
+    });
+
+    it("should dispatch RELEASE_REVISION_SUCCESS", () => {
+      const revision = {
+        architectures: ["amd64"],
+        revision: 3
+      };
+      const release = {
+        architecture: "amd64",
+        branch: null,
+        revision: 3,
+        risk: "edge",
+        track: "latest"
+      };
+      const store = mockStore({
+        options: {
+          snapName: "test",
+          csrfToken: "test",
+          defaultTrack: "latest"
+        },
+        pendingReleases: {
+          "3": {
+            revision: revision,
+            channels: ["latest/edge"]
+          }
+        },
+        pendingCloses: ["latest/edge"],
+        revisions: {
+          "3": revision
+        }
+      });
+
+      global.fetch = jest
+        .fn()
+        .mockResolvedValueOnce({
```

Comment:
Would be nice to have a comment before each `.mockResolvedOnce` explaining what API call is being mocked